### PR TITLE
Dave/stalwart 27

### PR DESCRIPTION
**Pull request by dave-augustus**
### :nut_and_bolt: Description: What code changed, and why?
The pipeline function is written to validate an unzip file.
The function basically checks if the unzip backup has organisations folder or not

### :chains: Related Resources
https://chefio.atlassian.net/jira/software/c/projects/STALWART/boards/361?modal=detail&selectedIssue=STALWART-27

### :athletic_shoe: How to Build and Test the Change

rebuild components/infra-proxy-service/

### :white_check_mark: Checklist

**All PRs** must tick these:

- [X] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [X] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [X] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [X] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [X] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [X] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [X] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [X] Tests added/updated? (all new code needs new tests)
- [X] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/infra-proxy-service/migrations/pipeline/utility_test.go [hunk 1]
```diff
@@ -129,3 +130,24 @@ func TestCreatePreview(t *testing.T) {
...
+
+	arg := args{
+		ctx:    context.Background(),
+		st:     &testDB.TestDB{},
+		mst:    &testDB.MigrationDB{},
+		result: pipeline.Result{Meta: pipeline.Meta{UnzipFolder: "../../testdata/backup", ServerID: "server1", MigrationID: "mig1"}},
+	}
+
+	res, err := ValidateZip(arg.ctx, arg.st, arg.mst, arg.result)
+	require.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (dave-augustus) [reply components/infra-proxy-service/migrations/pipeline/utility_test.go:150]: We are doing it already require.True(t, res.Meta.IsValid)

[the remark this answers is not in the document]
Comment (YashviJain01) [reply components/infra-proxy-service/migrations/pipeline/utility_test.go:150]: I think then require.True is only required.

[the remark this answers is not in the document]
Comment (dave-augustus) [reply components/infra-proxy-service/migrations/pipeline/utility_test.go:150]: Yes

**Review by YashviJain01 — APPROVED** [review]

File: components/infra-proxy-service/migrations/pipeline/utility.go [hunk 2]
```diff
@@ -341,3 +342,22 @@ func Unzip(ctx context.Context, mst storage.MigrationStorage, result pipeline.Re
 	}
 	return result, nil
 }
+
+func ValidateZip(ctx context.Context, st storage.Storage, mst storage.MigrationStorage, result pipeline.Result) (pipeline.Result, error) {
+	var err error
+	unzipFolder := result.Meta.UnzipFolder
+
+	_, err = os.Stat(unzipFolder + "/organizations")
```
Comment (kalroy) [inline components/infra-proxy-service/migrations/pipeline/utility.go:350]: We should be implementing it this way:
1. Find under unzip folder where the organisations folder exist. So if it is under <unzip_folder>/backup_20220221/new_backup/organisations
2. Set the location in results.Meta.UnzipFolder. Here it should be result.Meta.UnzipFolder = <unzip_folder>/backup_20220221/new_backup
3. Check key_dump.json in the same location so we expect key_dump.json under <unzip_folder>/backup_20220221/new_backup

**Review by kalroy — CHANGES_REQUESTED** [review]

**Review by kalroy — APPROVED** [review]

**Comment by sonali523** [comment]
@kalroy In this issue, should we check for members.json and admins.json?
At least empty files should be there

**Pull request merged**